hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -334,22 +337,43 @@ public HoodieTableConfig() {
     super();
   }
 
-  private void fetchConfigs(FileSystem fs, String metaPath) throws IOException {
+  private static TypedProperties fetchConfigs(FileSystem fs, String metaPath) throws IOException {
     Path cfgPath = new Path(metaPath, HOODIE_PROPERTIES_FILE);
-    try (FSDataInputStream is = fs.open(cfgPath)) {
-      props.load(is);
-    } catch (IOException ioe) {
-      if (!fs.exists(cfgPath)) {
-        LOG.warn("Run `table recover-configs` if config update/delete failed midway. Falling back to backed up configs.");
-        // try the backup. this way no query ever fails if update fails midway.
-        Path backupCfgPath = new Path(metaPath, HOODIE_PROPERTIES_FILE_BACKUP);
-        try (FSDataInputStream is = fs.open(backupCfgPath)) {
+    Path backupCfgPath = new Path(metaPath, HOODIE_PROPERTIES_FILE_BACKUP);
+    int readRetryCount = 0;
+    boolean found = false;
+
+    TypedProperties props = new TypedProperties();
+    while (readRetryCount++ < MAX_READ_RETRIES) {
+      for (Path path : Arrays.asList(cfgPath, backupCfgPath)) {
+        // Read the properties and validate that it is a valid file
+        try (FSDataInputStream is = fs.open(path)) {
+          props.clear();
           props.load(is);
+          found = true;
+          ValidationUtils.checkArgument(validateChecksum(props));
```

Comment:
I think this LOC would break backward compatibility.

If a Hudi table was created with a much older Hudi version (say 0.11.0) and read by 0.14.0, then this LOC would throw IAE because checksum property doesn't exist in older Hudi. 

Maybe we should check if the checksum exists first. wdyt? @prashantwason

Reply:
@CTTY Agree.
See the issue [issue#10404](https://github.com/apache/hudi/issues/10404).